internal/driver/internal/k8s/logicalvolume_service.go

```diff
@@ -313,36 +312,39 @@ func (s *LogicalVolumeService) ExpandVolume(ctx context.Context, volumeID string
 		return err
 	}
 
-	// wait until topolvm-node expands the target volume
-	for {
-		logger.Info("waiting for update of 'status.currentSize'", "name", lv.Name)
-		select {
-		case <-ctx.Done():
-			return ctx.Err()
-		case <-time.After(1 * time.Second):
-		}
-
+	return wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
```

Comment:
I agree actually. I dont think the k8s libraries have a way to specify "no step limit though" so Im just gonna run with MAX_INT. Alternatively we could set this to an arbitrarily high number so that we avoid leaks in case the context is never closed (highly unlikely but possible)